Mathlib/Analysis/NormedSpace/Alternating/Basic.lean

```diff
@@ -0,0 +1,601 @@
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sébastien Gouëzel, Yury Kudryashov, Heather Macbeth, Patrick Massot
+-/
+import Mathlib.Topology.Algebra.Module.Alternating.Topology
+import Mathlib.Analysis.NormedSpace.Multilinear.Basic
+
+/-!
+# Operator norm on the space of continuous alternating maps
+
+In this file we show that continuous alternating maps
+from a seminormed space to a (semi)normed space form a (semi)normed space.
+We also prove basic facts about this norm
+and define bundled versions of some operations on continuous alternating maps.
+
+Most proofs just invoke the corresponding fact about continuous multilinear maps.
+-/
+
+noncomputable section
+
+open scoped BigOperators NNReal
+open Finset Metric
+
+/-!
+### Type variables
+
+We use the following type variables in this file:
+
+* `𝕜` : a nontrivially normed field;
+* `ι`: a finite index type with decidable equality;
+* `E`, `F`, `G`: (semi)normed vector spaces over `𝕜`.
+-/
+
+/-- Applying a continuous alternating map to a vector is continuous in both coordinates. -/
+instance ContinuousAlternatingMap.instContinuousEval {𝕜 ι E F : Type*}
+    [NormedField 𝕜] [Finite ι] [SeminormedAddCommGroup E] [NormedSpace 𝕜 E]
+    [TopologicalSpace F] [AddCommGroup F] [IsTopologicalAddGroup F] [Module 𝕜 F] :
+    ContinuousEval (E [⋀^ι]→L[𝕜] F) (ι → E) F :=
+  .of_continuous_forget continuous_toContinuousMultilinearMap
+
+section Seminorm
+
+universe u wE wF wG v
+variable {𝕜 : Type u} {n : ℕ} {E : Type wE} {F : Type wF} {G : Type wG} {ι : Type v}
+  [NontriviallyNormedField 𝕜]
+  [SeminormedAddCommGroup E] [NormedSpace 𝕜 E]
+  [SeminormedAddCommGroup F] [NormedSpace 𝕜 F]
+  [SeminormedAddCommGroup G] [NormedSpace 𝕜 G]
+
+/-!
+### Continuity properties of alternating maps
+
+We relate continuity of alternating maps to the inequality `‖f m‖ ≤ C * ∏ i, ‖m i‖`, in
+both directions. Along the way, we prove useful bounds on the difference `‖f m₁ - f m₂‖`.
+-/
+namespace AlternatingMap
+
+/-- If `f` is a continuous alternating map on `E`
+and `m` is an element of `ι → E` such that one of the `m i` has norm `0`, then `f m` has norm `0`.
+
+Note that we cannot drop the continuity assumption because `f (m : Unit → E) = f (m ())`,
+where the domain has zero norm and the codomain has a nonzero norm
+does not satisfy this condition. -/
+nonrec theorem norm_map_coord_zero (f : E [⋀^ι]→ₗ[𝕜] F) (hf : Continuous f)
+    {m : ι → E} {i : ι} (hi : ‖m i‖ = 0) : ‖f m‖ = 0 :=
+  f.norm_map_coord_zero hf hi
+
+variable [Fintype ι]
+
+/-- If an alternating map in finitely many variables on seminormed spaces
+sends vectors with a component of norm zero to vectors of norm zero
+and satisfies the inequality `‖f m‖ ≤ C * ∏ i, ‖m i‖` on a shell `ε i / ‖c i‖ < ‖m i‖ < ε i`
+for some positive numbers `ε i` and elements `c i : 𝕜`, `1 < ‖c i‖`,
+then it satisfies this inequality for all `m`.
+
+The first assumption is automatically satisfied on normed spaces, see `bound_of_shell` below.
```

Comment:
This should be "The second assumption", right?

Reply:
No, the first assumption `hf0` is automatically satisfied for normed spaces.